tests/test_hbavss.py

```diff
@@ -54,18 +54,24 @@ def get_avss_params(n, t):
     crs = gen_pc_const_crs(t, g=g, h=h)
 
     values = [ZR.random()] * (t+1)
-    avss_tasks = [None]*n
+    avss_tasks = [None] * n
     dealer_id = randint(0, n-1)
 
+    shares = [None] * n
     with ExitStack() as stack:
+        hbavss_list = []
         for i in range(n):
             hbavss = HbAvssBatch(pks, sks[i], crs, n, t, i, sends[i], recvs[i])
+            hbavss_list.append(hbavss)
             stack.enter_context(hbavss)
             if i == dealer_id:
                 avss_tasks[i] = asyncio.create_task(hbavss.avss(0, values=values))
             else:
                 avss_tasks[i] = asyncio.create_task(hbavss.avss(0, dealer_id=dealer_id))
-        shares = await asyncio.gather(*avss_tasks)
+        for i in range(n):
+            shares[i] = await hbavss_list[i].shares_future
```

Comment:
use `asyncio.gather` this serializes the code.

Reply:
The batch avss doesn't return anything now. It output the value in a Future variable.

Reply:
You can collect all futures and then do a gather on them:
```
share_futures = await asyncio.gather(*[hbavss_list[i].shares_future for i in range(n)])
```

Reply:
Yeah, this is a good way.